expressions/predicate/ComparisonPredicate.cpp

```diff
@@ -127,7 +127,9 @@ TupleIdSequence* ComparisonPredicate::getAllMatches(
 
   if (fast_comparator_.get() == nullptr) {
     return GenerateSequenceForStaticResult(accessor, filter, existence_map, static_result_);
-  } else if (sub_blocks_ref != nullptr) {
+  } else if (sub_blocks_ref != nullptr && comparison_.isBasicComparison()) {
+    // NOTE(Jianqiao): sub-block indices only apply to basic comparisons.
```

Comment:
Perhaps it is not a single `TODO` but requires co-changes in many places. So let's leave it here. Later if there is a need for the indexing support for non-trivial comparisons, then this place will be naturally updated.